x/sequencers/keeper/msg_server.go

```diff
@@ -0,0 +1,112 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/sequencers/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+func (m msgServer) CreateSequencer(goCtx context.Context, msg *types.MsgCreateSequencer) (*types.MsgCreateSequencerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	accAddr := msg.MustAccAddr() // ensured in validate basic
+	allow, err := m.IsSigned(ctx, accAddr, msg.GetKeyAndSig(), msg.GetPayload())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "check sig")
+	}
+	if !allow {
+		return nil, gerrc.ErrUnauthenticated
+	}
+	operator := msg.MustOperatorAddr() // checked in validate basic
+	if _, ok := m.GetSequencer(ctx, operator); ok {
+		return nil, gerrc.ErrAlreadyExists
+	}
+
+	v := msg.GetKeyAndSig().Validator()
+	v.OperatorAddress = operator.String()
+	m.SetSequencer(ctx, v)
```

Comment:
why u dont get the Validator from `msg` instead of `msg.GetKeyAndSig()`?
than u can use `NewValidator` to have a valid validtor object

Reply:
what's the benefit?